src/_igraph/convert.c

```diff
@@ -2606,11 +2676,120 @@ int igraphmodule_PyObject_to_vector_int_ptr_t(PyObject* list, igraph_vector_ptr_
     if (igraph_vector_ptr_push_back(vec, subvec)) {
       Py_DECREF(it);
       igraph_vector_int_destroy(subvec);
+      free(subvec);
       igraph_vector_ptr_destroy_all(vec);
       return 1;
     }
 
-    /* ownership of 'subvec' taken by 'vec' here */
+    /* ownership of 'subvec' taken by 'vec' here, we should still free the pointer */
+  }
+
+  Py_DECREF(it);
+  return 0;
+}
+
+/**
+ * \ingroup python_interface_conversion
+ * \brief Converts a Python list of lists to an \c igraph_vector_list_t
+ *        (containing \c igraph_vector_int_t items).
```

Comment:
An `igraph_vector_list_t` contains `igraph_vector_t` items; the comment should be fixed. The code is OK.